src/services/dashboards/dashboard-customize/modules/VariableMoreButtonDropdown.vue

```diff
@@ -118,18 +132,34 @@ const handleOpenOverlay = () => {
 };
 const handleClickButton = () => {
     if (visibleMenu.value) {
-        hideContextMenu();
-        // Reflect selectedForUpdate changes after the dropdown is closed.
-        updateVariablesUse();
+        handleHideMenu();
     } else {
         state.selectedForUpdate = state.selected;
-        showContextMenu(true); // update reorderedMenu automatically
+        showContextMenu(); // update reorderedMenu automatically
     }
 };
+
+const handleHideMenu = () => {
+    hideContextMenu();
+    // Reflect selectedForUpdate changes after the dropdown is closed.
+    updateVariablesUse();
+};
+
 const handleSelectVariable = (changedSelected: MenuItem[]) => {
     state.selectedForUpdate = changedSelected;
 };
 
+const handleUpdateSearchText = debounce((text: string) => {
+    searchText.value = text;
+    reloadMenu();
+}, 200);
+
+watch(visibleMenu, (_visibleMenu) => {
+    if (_visibleMenu) {
+        initiateMenu();
+    }
+}, { immediate: true });
```

Comment:
nitpick: I think `visibleMenu.value` should be watched!

```suggestion
watch(visibleMenu.value, (_visibleMenu) => {
    if (_visibleMenu) {
        initiateMenu();
    }
}, { immediate: true });
```

Reply:
No this is correct..